java/client/src/main/java/glide/api/commands/StreamBaseCommands.java

```diff
@@ -512,4 +514,121 @@ CompletableFuture<Map<String, Map<String, String[][]>>> xreadgroup(
      * </pre>
      */
     CompletableFuture<Long> xack(String key, String group, String[] ids);
+
+    /**
+     * Returns stream message summary information for pending messages matching a given range of IDs.
+     *
+     * @see <a href="https://valkey.io/commands/xpending/">valkey.io</a> for details.
+     * @param key The key of the stream.
+     * @param group The consumer group name.
+     * @return An array that includes the summary of pending messages, with the format
+     * <code>[NumOfMessages, StartId, EndId, [Consumer, NumOfMessages]]</code>, where:
+     * <ul>
+     *      <li> <code>NumOfMessages</code>: The total number of pending messages for this consumer group.
+     *      <li> <code>StartId</code>: The smallest ID among the pending messages.
+     *      <li> <code>EndId</code>: The greatest ID among the pending messages.
+     *      <li> <code>[Consumer, NumOfMessages]</code>: An array of every consumer in the consumer group with at least one pending message, and the number of pending messages it has.
```

Comment:
Should we convert this to a map?
If no, please emphasize that it is an 2D array - it is a bit unclear.

Reply:
I will emphasize this is a 2D array.